test/widget_tests/pre_auth_screens/login_test.dart

```diff
@@ -0,0 +1,146 @@
+      },
+      child: const Text('ravidi'),
+    ),
+    navigatorKey: locator<NavigationService>().navigatorKey,
+    onGenerateRoute: router.generateRoute,
+  );
+}
+
+Future<void> showLoginScreen(tester) async {
+  await tester.pumpWidget(createApp());
+  await tester.pump();
+  await tester.tap(find.textContaining('ravidi'));
+  await tester.pumpAndSettle();
+}
+
+void main() {
+  SizeConfig().test();
+  testSetupLocator();
+  group('Test for Login screen', () {
+    testWidgets('Check if Login screen shows up', (tester) async {
+      await tester.pumpWidget(createApp());
+      await tester.pump();
+
+      expect(find.byType(Login), findsNothing);
+
+      await tester.tap(find.textContaining('ravidi'));
+      await tester.pumpAndSettle();
+
+      expect(find.byType(Login), findsOneWidget);
+    });
+
+    testWidgets('Check if back button works', (tester) async {
+      await tester.pumpWidget(createApp());
+      await tester.pump();
+
+      expect(find.byType(Login), findsNothing);
+
+      await tester.tap(find.textContaining('ravidi'));
+      await tester.pumpAndSettle();
+
+      expect(find.byType(Login), findsOneWidget);
+
+      await tester.tap(find.descendant(
+        of: find.byType(AppBar),
+        matching: find.byType(IconButton),
+      ));
+      await tester.pumpAndSettle();
+
+      expect(find.byType(Login), findsNothing);
+    });
+
+    testWidgets('Check if all widgets are present', (tester) async {
+      await showLoginScreen(tester);
+
+      expect(find.byType(Scaffold), findsOneWidget);
+      expect(find.byType(AppBar), findsOneWidget);
+      expect(find.byType(SingleChildScrollView), findsOneWidget);
+      expect(find.byType(CustomRichText), findsOneWidget);
+      expect(find.byType(TextFormField), findsNWidgets(2));
+      expect(find.byType(RaisedRoundedButton), findsOneWidget);
+    });
+
+    testWidgets('Check if all texts show up properly', (tester) async {
+      await showLoginScreen(tester);
+
+      // expect(find.text("We're Glad you're Back "), findsOneWidget);
+      expect(
+          find.textContaining("Enter your registered Email"), findsOneWidget);
+      expect(find.textContaining("Enter your password"), findsOneWidget);
+      expect(find.textContaining("Forgot password"), findsOneWidget);
+    });
+
+    testWidgets('Check if entering text in TextField(s) works', (tester) async {
+      await showLoginScreen(tester);
+
+      final emailField = tester.firstWidget(find.byType(TextFormField).first);
+      final passwordField =
+          tester.widgetList(find.byType(TextFormField).last).first;
+
+      await tester.enterText(find.byType(TextFormField).first, 'ravidi');
+      await tester.enterText(find.byType(TextFormField).last, 'shaikh');
+
+      expect((emailField as TextFormField).controller!.text, 'ravidi');
+      expect((passwordField as TextFormField).controller!.text, 'shaikh');
+    });
+
+    // TODO: uncomment after implementing mock LoginViewModel
```

Comment:
Shouldn't implementing this be a part of this PR? Please include it so that we have a comprehensive test. Create an issue to cover this if necessary and include it in this PR. That way login testing is more complete.

Reply:
Actually I'm currently working on widget and screen tests and so I've a complete understanding of the widget code. But for the model ones, I've not dived too deep. So if possible, merge this PR for time being so that I can close #986. And then I'll start working on model tests, and implement this mock for sure.